TypeCobol/Compiler/Domain/ProgramSymbolTableBuilder.cs

```diff
@@ -246,6 +246,74 @@ public override void EndLinkageSection()
             CurrentDataDivisionSection = null;
         }
 
+        /// <summary>
+        /// Resolve in the current program linkage section, the given parameter.
+        /// </summary>
+        /// <param name="p">The parameter to resolve.</param>
+        /// <returns>The resolved variable if any, null otherwise</returns>
+        private VariableSymbol ResolveUsingParameter(CallTargetParameter p)
+        {
+            System.Diagnostics.Debug.Assert(p.StorageArea != null);
+            System.Diagnostics.Debug.Assert(p.StorageArea.SymbolReference != null);
+            string parameterName = p.StorageArea.SymbolReference.Name;
+            var parameterCandidateSymbols = this.CurrentProgram.LinkageData.Lookup(parameterName);
+            if (parameterCandidateSymbols != null && parameterCandidateSymbols.Count == 1)
+            {
+                var parameterSymbol = parameterCandidateSymbols.Symbol;
+                if (p.SharingMode == null)
+                    parameterSymbol.SetFlag(Symbol.Flags.ByReference, true);
+                else
+                    switch (p.SharingMode.Value)
+                    {
+                        case ParameterSharingMode.ByContent:
+                            parameterSymbol.SetFlag(Symbol.Flags.ByContent, true);
+                            break;
+                        case ParameterSharingMode.ByReference:
+                            parameterSymbol.SetFlag(Symbol.Flags.ByReference, true);
+                            break;
+                        case ParameterSharingMode.ByValue:
+                            parameterSymbol.SetFlag(Symbol.Flags.ByValue, true);
+                            break;
+                    }
+
+                return parameterSymbol;
+            }
+
+            return null;
+        }
+
+        /// <summary>
+        /// Starting a Program PROCEDURE DIVISION => Collect all parameters.
+        /// </summary>
+        /// <param name="header"></param>
+        public override void StartProcedureDivision(ProcedureDivisionHeader header)
```

Comment:
Same remarks here, this method must be removed.

Reply:
Done